### ISSUE TITLE
Agent does not starts due to missing capability "CAP_CAP_CHOWN"

### ISSUE THREAD

**Issue by lenaing** [aws/amazon-ecs-agent#3471]
### Summary

- OS : CentOS Linux release 7.9.2009 (Core)
- RPM : amazon-ecs-init-1.65.1-1.x86_64
- Docker version 19.03.8, build afacb8b

### Description

Agent does not start, with the following error : 

```
[ERROR] could not start Agent: API error (500): {"message":"linux spec capabilities: Unknown capability to add: \"CAP_CAP_CHOWN\""}
``` 

### Expected Behavior

Agent starts.

### Observed Behavior

```
[root@example ~]$ cat /proc/17042/status | grep Cap # This is the docker daemon
CapInh: 0000000000000000
CapPrm: 0000001fffffffff
CapEff: 0000001fffffffff
CapBnd: 0000001fffffffff
CapAmb: 0000000000000000
[root@example ~]$ capsh --decode=0000001fffffffff
0x0000001fffffffff=cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,35,36
[root@example ~]$ capsh --print
Current: = cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,35,36+ep
Bounding set =cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,35,36
Securebits: 00/0x0/1'b0
 secure-noroot: no (unlocked)
 secure-no-suid-fixup: no (unlocked)
 secure-keep-caps: no (unlocked)
uid=0(root)
gid=0(root)
groups=0(root)
```

I believe this was introduced by https://github.com/aws/amazon-ecs-agent/commit/0f00378c7da77c0c8b5e24a5ad1ce974a2b66f94#diff-070574e949d062f9970c3859c2c2c4a26f9a6f816d3502d81764a6e8889e2283R86 maybe just `CHOWN` was intended ?

I'm not SYSCAP fluent though, so the issue may lie elswhere.

**Comment by ubhattacharjya** [comment]
Hi @lenaing ,

Sorry to hear that you are facing this issue. It would be helpful to know how you are starting ECS agent so that we could replicate the issue at our end.

Thanks,
Utsa

**Comment by lenaing** [comment]
Hi,

The agent is managed by the RPM and launched through _systemd_ following this document : https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-install.html#ecs-agent-install-nonamazonlinux

Here is the full log : 

```
Nov 08 15:34:13 example systemd[1]: Starting Amazon Elastic Container Service - container agent...
Nov 08 15:34:13 example amazon-ecs-init[19910]: 2022-11-08T14:34:13Z [INFO] pre-start
Nov 08 15:34:13 example amazon-ecs-init[19910]: 2022-11-08T14:34:13Z [INFO] pre-start: enabling loopback routing
Nov 08 15:34:13 example amazon-ecs-init[19910]: 2022-11-08T14:34:13Z [INFO] pre-start: disabling ipv6 router advertisements
Nov 08 15:34:13 example amazon-ecs-init[19910]: 2022-11-08T14:34:13Z [INFO] pre-start: creating credentials proxy route
Nov 08 15:34:13 example amazon-ecs-init[19910]: 2022-11-08T14:34:13Z [INFO] pre-start: checking ecs agent container image loaded presence
Nov 08 15:34:13 example amazon-ecs-init[19910]: 2022-11-08T14:34:13Z [INFO] pre-start: ecs agent container image loaded presence: true
Nov 08 15:34:13 example amazon-ecs-init[19910]:
Nov 08 15:34:13 example systemd[1]: Started Amazon Elastic Container Service - container agent.
Nov 08 15:34:13 example amazon-ecs-init[19924]: 2022-11-08T14:34:13Z [INFO] start
Nov 08 15:34:13 example amazon-ecs-init[19924]: 2022-11-08T14:34:13Z [INFO] Container name: /ecs-agent
Nov 08 15:34:13 example amazon-ecs-init[19924]: 2022-11-08T14:34:13Z [INFO] Removing existing agent container ID: 59fbc12920d91b0045fb37e14fe70ef21b303fe68559ebc598456870db9b80c2
Nov 08 15:34:13 example amazon-ecs-init[19924]: 2022-11-08T14:34:13Z [INFO] Starting Amazon Elastic Container Service Agent
Nov 08 15:34:13 example amazon-ecs-init[19924]: 2022-11-08T14:34:13Z [ERROR] could not start Agent: API error (500): {"message":"linux spec capabilities: Unknown capability to add: \"CAP_CAP_CHOWN\""}
Nov 08 15:34:13 example amazon-ecs-init[19924]:
Nov 08 15:34:13 example systemd[1]: ecs.service: main process exited, code=exited, status=255/n/a
Nov 08 15:34:13 example amazon-ecs-init[19933]: 2022-11-08T14:34:13Z [INFO] post-stop
Nov 08 15:34:13 example amazon-ecs-init[19933]: 2022-11-08T14:34:13Z [INFO] Cleaning up the credentials endpoint setup for Amazon Elastic Container Service Agent
Nov 08 15:34:13 example amazon-ecs-init[19933]:
Nov 08 15:34:13 example systemd[1]: Unit ecs.service entered failed state.
Nov 08 15:34:13 example systemd[1]: ecs.service failed.
```

Regards,

**Comment by itshili** [comment]
Hi there, 

It appears there is compatibility issue with the newly released "ecs-init version: 1.65.1-1" and older docker versions (Docker: 18.06.1-ce & 19.03.6-ce)

Looking at the documentation page for [Installing the Amazon ECS container agent on a non-Amazon Linux EC2 instance](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-install.html#ecs-agent-install-nonamazonlinux), it advised for customers to use the lastest version of docker and validate that meets the minimum version requirement. 

Steps: 
    3. Install the latest version of Docker on your instance.
    4. Check your Docker version to verify that your system meets the minimum version requirement.

Reviewing the documentation for [Amazon ECS-Optimized Amazon Linux 2 AMI container agent versions](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-versions.html#al2-optimized-ami-agent-versions), we can see the following note: 

>   **Note**   
>   `As new Amazon ECS-optimized Amazon Linux 2 AMIs and Amazon ECS agent versions are released, older versions are still available for launch in Amazon EC2. However, we encourage you to update to the latest version of the Amazon ECS agent and to keep your container instance software up to date. If you request support for an older version of the Amazon ECS agent through AWS Support, you may be asked to move to the latest version as a part of the support process.`

    Amazon ECS-optimized Amazon Linux 2 AMI: 20221102
    Amazon ECS container agent version: 1.65.1
    Docker version: 20.10.17
    ecs-init version: 1.65.1-1


Therefore, I would recommend updating the docker version to `Docker version: 20.10.17` if you want to use `ecs-init version: 1.65.1-1`. I've tested it and it works.

**Comment by sparrc** [comment]
Hi all, we are aware of this issue and working to fix it. It will be fixed in agent/init version 1.66.1 which we are currently working to build and publish ASAP.

Current mitigation options:

1. upgrade docker to 20.x:

```
sudo yum upgrade -y docker
```
2. pgrade to launch the latest ECS-Optimized AMI. The latest AMI ID can be retrieved from SSM parameters: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/retrieve-ecs-optimized_AMI.html

**Comment by ubhattacharjya** [comment]
Hi,

The fix has been deployed with ECS agent 1.66.2 .

Kindly upgrade to use the latest ECS Optimized AMI.

Thanks
